docs/data/data-grid/recipes-row-grouping/RowGroupingToolbar.tsx

```diff
@@ -0,0 +1,190 @@
+    (field: string) => (event: React.DragEvent<HTMLDivElement>) => {
+      setDraggedChip(field);
+      event.dataTransfer.effectAllowed = 'move';
+    };
+
+  const handleChipDragEnd = () => {
+    setDraggedChip(null);
+  };
+
+  const handleChipDragOver =
+    (targetField: string) => (event: React.DragEvent<HTMLDivElement>) => {
+      event.preventDefault();
+
+      const draggedField = draggedChip || draggedColumn;
+      if (!draggedField || draggedField === targetField) {
+        return;
+      }
+
+      const currentIndex = rowGroupingModel.indexOf(draggedField);
+      const targetIndex = rowGroupingModel.indexOf(targetField);
+
+      if (currentIndex === -1 || targetIndex === -1) {
+        return;
+      }
+
+      if (currentIndex !== targetIndex) {
+        const newModel = [...rowGroupingModel];
+        newModel.splice(currentIndex, 1);
+        newModel.splice(targetIndex, 0, draggedField);
+        apiRef.current.setRowGroupingModel(newModel);
+      }
+    };
+
+  const moveRowGroup = (field: string, position: number) => {
+    if (position < 0 || position > rowGroupingModel.length) {
+      return;
+    }
+
+    const currentIndex = rowGroupingModel.indexOf(field);
+    const newModel = [...rowGroupingModel];
+    newModel.splice(currentIndex, 1);
+    newModel.splice(position, 0, field);
+    apiRef.current.setRowGroupingModel(newModel);
+  };
+
+  const removeRowGroup = (field: string) => {
+    apiRef.current.removeRowGroupingCriteria(field);
+  };
+
+  return (
+    <StyledToolbar
+      aria-label="Row grouping"
+      onDragOver={handleToolbarDragOver}
+      onDragLeave={handleToolbarDragLeave}
+    >
+      <MoveUpIcon fontSize="small" color="action" sx={{ mr: 1 }} />
+      {rowGroupingModel.length > 0 ? (
+        <React.Fragment>
+          {rowGroupingModel.map((field, index) => {
+            const isDraggedField = draggedChip === field || draggedColumn === field;
+            return (
+              <React.Fragment key={field}>
+                {index > 0 && <ChevronRightIcon fontSize="small" color="action" />}
+                <ToolbarButton
+                  id={field}
+                  render={({
+                    children,
+                    color,
+                    size,
+                    ref,
+                    onKeyDown,
+                    ...chipProps
+                  }) => {
+                    const handleKeyDown = (
+                      event: React.KeyboardEvent<HTMLDivElement>,
+                    ) => {
+                      if (event.key === 'ArrowRight' && event.shiftKey) {
+                        moveRowGroup(field, index + 1);
+                      } else if (event.key === 'ArrowLeft' && event.shiftKey) {
+                        moveRowGroup(field, index - 1);
+                      } else {
+                        onKeyDown?.(event);
+                      }
+                    };
+
+                    return (
+                      <Chip
```

Comment:
Should we disable these chips when pivot mode is active? They won't work anyway.